web/packages/extension/js/lv0.js

```diff
@@ -141,26 +141,40 @@ get_sync_storage(["ruffle_enable", "ignore_optout"], function (data) {
-        let scriptelem = document.createElement("script");
-        setup_scriptelem.innerHTML = setup_src;
-        /* The setup_scriptelem sets runtime_path & obfuscated_event_prefix *
-         * and runs the plugin polyfill, which must run before any flash    *
-         * detection scripts.                                               */
         (document.head || document.documentElement).appendChild(
-            setup_scriptelem
+            polyfill_script
         );
-        scriptelem.src = ext_path + "dist/ruffle.js";
-        (document.head || document.documentElement).appendChild(scriptelem);
+
+        // Load Ruffle script asynchronously. By doing so, we can inject extra variables and isolate them from the global scope.
+        (async function () {
+            let ruffle_src_resp = await fetch(ext_path + "dist/ruffle.js");
+            if (ruffle_src_resp.ok) {
+                let ruffle_source =
+                    '(function () { var ruffleRuntimePath = "' +
+                    ext_path +
+                    '";\nvar obfuscated_event_prefix = "' +
+                    obfuscated_event_prefix +
+                    '";\n' +
+                    (await ruffle_src_resp.text()) +
+                    "}())";
+                let ruffle_script = document.createElement("script");
+                ruffle_script.appendChild(
+                    document.createTextNode(ruffle_source)
+                );
```

Comment:
Might as well make all of these use `camelCase` naming while we're modifying it

Reply:
Yeah, I shouldn't have been that lazy. :D I've now updated all the variables / functions that don't rely on other files.